### PR TITLE
Rename old data refresh to `legacy_data_refresh`

### LINKED ISSUE #4146: Create the new data refresh DAG factory and move initial steps into Airflow
## Problem

This issue tracks the creation of a new data refresh DAG factory to generate the new data refresh DAGs that will not rely on the ingestion server, and moving the initial steps (described below) into the DAGs. At the end of this step the DAGs will _not_ be functional/able to run a full refresh.

## Description

We’ll create a new data refresh DAG factory to generate data refresh DAGs for each existing media_type and environment. Currently these four will be generated:

- staging_audio_data_refresh
- staging_image_data_refresh
- production_audio_data_refresh
- production_image_data_refresh

Because the environment is added as a prefix, there will be no collision with the existing DAG ids. In this initial step, we we will add only a small portion of the logic in order to make the PR easier to review. The first steps are already implemented in the current data refresh and can simply be copied:

- Get the current record count from the target API table; this must be modified to take the environment as an argument
- Perform concurrency checks on the other data refreshes and conflicting DAGs; this must be modified to include the now larger list of data refresh DAG ids
- Get the name of the Elasticsearch index currently mapped to the target_alias
- Generate the new index suffix

We will include new tasks to perform the initial few steps of the ingestion server’s work:

- Copy Data: this should be a TaskGroup that will have multiple tasks for creating the FDW from the upstream DB to the downstream DB, running the copy_data query, and so on. It should fully replace the implementation of [refresh_api_table](https://github.com/WordPress/openverse/blob/05ff48d05f2163104151c5589cf352a156bc6a97/ingestion_server/ingestion_server/ingest.py#L248) in the ingestion server. All steps in this section are SQL queries that can be implemented using the existing [PostgresHook and PGExecuteQueryOperator](https://github.com/WordPress/openverse/blob/05ff48d05f2163104151c5589cf352a156bc6a97/catalog/dags/common/sql.py).
- Create Index: we can use our existing [Elasticsearch tasks](https://github.com/WordPress/openverse/blob/05ff48d05f2163104151c5589cf352a156bc6a97/catalog/dags/common/elasticsearch.py#L82) to create the new elasticsearch index with the index suffix generated in the previous task.

## Additional context

See [this section](https://docs.openverse.org/projects/proposals/ingestion_server_removal/20240328-implementation_plan_ingestion_server_removal.html#create-the-new-data-refresh-dag-factory-and-move-initial-steps-into-airflow) of the IP

### PR DESCRIPTION
**Pull request by stacimc**
## Fixes

Related to #4146 by @stacimc 

## Description

This PR makes no changes to the current data refreshes, but renames the _directory_ to `legacy_data_refresh` to avoid naming collisions. I think this is the easiest path forward for simply deleting it when the new DAGs are running.

This is pulled into a separate PR because the diff for adding the new data refresh would otherwise be extremely confusing.

## Testing Instructions

Locally, run an image and an audio DAG for a few minutes to ingest some new records (I ran Cleveland and Jamendo). Now run the (regular) `audio_data_refresh` and `image_data_refresh` locally and verify that they still work. Note that the reported record count will have increased because you got some new local data before running them.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: catalog/dags/elasticsearch_cluster/healthcheck_dag.py [hunk 1]
```diff
@@ -24,12 +24,12 @@
 from airflow.exceptions import AirflowSkipException
 from airflow.providers.elasticsearch.hooks.elasticsearch import ElasticsearchPythonHook
 from elasticsearch import Elasticsearch
+from legacy_data_refresh.data_refresh_types import DATA_REFRESH_CONFIGS
```
Comment (AetherUnbound) [inline catalog/dags/elasticsearch_cluster/healthcheck_dag.py:27]: I believe we'll need to add the new folder to the list of first party modules in our isort/rust config so they still show up in the group below.

**Review by dhruvkb — APPROVED** [review]
Makes sense 👍!

**Comment by stacimc** [comment]
Hm, that's interesting @obulat -- what DAGs had you run, on top of the sample data? I don't see that locally, and it seems like it would be an issue with the _data_ rather than the DAG.

From `varchar(2000)` it looks like possibly a creator/creator url/title that's too long? But the same constraints are on the catalog tables, so that's quite odd... Do you mind repeating your tests on `main`?

**Comment by obulat** [comment]
> Hm, that's interesting @obulat -- what DAGs had you run, on top of the sample data? I don't see that locally, and it seems like it would be an issue with the _data_ rather than the DAG.
> 
> From `varchar(2000)` it looks like possibly a creator/creator url/title that's too long? But the same constraints are on the catalog tables, so that's quite odd... Do you mind repeating your tests on `main`?

I found the problem. It's a wikimedia title for item https://commons.wikimedia.org/w/index.php?curid=148076183:
```
<div class='fn'> <div style='font-size:0.9em;display:inline-block;'>French: <div style='display:inline-block' dir='ltr' lang='fr'><i>Rustican <span class='mw-valign-text-top' typeof='mw:File/Frameless'><a href='https://www.wikidata.org/wiki/Q123991443#P1476' title='Edit this at Wikidata'><img alt='Edit this at Wikidata' src='https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/10px-OOjs_UI_icon_edit-ltr-progressive.svg.png' decoding='async' width='10' height='10' class='mw-file-element' srcset='https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/15px-OOjs_UI_icon_edit-ltr-progressive.svg.png 1.5x, https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/20px-OOjs_UI_icon_edit-ltr-progressive.svg.png 2x' data-file-width='20' data-file-height='20'></a></span></i></div></div><div style='display: none;'>title QS:P1476,fr:'Rustican <span class='mw-valign-text-top' typeof='mw:File/Frameless'><a href='https://www.wikidata.org/wiki/Q123991443#P1476' title='Edit this at Wikidata'><img alt='Edit this at Wikidata' src='https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/10px-OOjs_UI_icon_edit-ltr-progressive.svg.png' decoding='async' width='10' height='10' class='mw-file-element' srcset='https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/15px-OOjs_UI_icon_edit-ltr-progressive.svg.png 1.5x, https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/20px-OOjs_UI_icon_edit-ltr-progressive.svg.png 2x' data-file-width='20' data-file-height='20'></a></span>'</div> <div style='display: none;'>label QS:Lfr,'Rustican <span class='mw-valign-text-top' typeof='mw:File/Frameless'><a href='https://www.wikidata.org/wiki/Q123991443#P1476' title='Edit this at Wikidata'><img alt='Edit this at Wikidata' src='https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/10px-OOjs_UI_icon_edit-ltr-progressive.svg.png' decoding='async' width='10' height='10' class='mw-file-element' srcset='https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/15px-OOjs_UI_icon_edit-ltr-progressive.svg.png 1.5x, https://upload.wikimedia.org/wikipedia/commons/thumb/8/8a/OOjs_UI_icon_edit-ltr-progressive.svg/20px-OOjs_UI_icon_edit-ltr-progressive.svg.png 2x' data-file-width='20' data-file-height='20'></a></span>'</div></div>
```

In catalog, the title has max length set to 5000, and in the API it's 2000. I wonder how that never caused problems for us?
It seems that we need to escape the wikimedia titles. 

Looks like we need to open a new issue for this, it's definitely not a problem with this PR.

**Comment by stacimc** [comment]
Wow, what a find @obulat!

**Pull request merged**